app/presenters/curation_concerns/member_presenter_factory_override.rb

```diff
@@ -6,3 +6,24 @@
 # GenericWork#member_presenters. This is actually a custom CHF one despite the
 # namespace.
 CurationConcerns::MemberPresenterFactory.work_presenter_class = CurationConcerns::GenericWorkShowPresenter
+
+# These overrides are taken from Plum's EfficientMemberPresenterFactory,
+# https://github.com/pulibrary/plum/blob/f979e6688e7ddac446c1747fc0ee986aeb150869/app/factories/efficient_member_presenter_factory.rb
+CurationConcerns::MemberPresenterFactory.class_eval do
+  # Override to get all member solr docs in one solr query, rather than getting them one at a time in a presenter factory
+  def member_presenters(ids = ordered_ids, presenter_class = composite_presenter_class)
+    @member_presenters ||=
+      begin
+        ordered_docs.select { |x| ids.include?(x.id) }.map do |doc|
+          presenter_class.new(doc, *presenter_factory_arguments)
+        end
+      end
+  end
+
```

Comment:
I am not a fan of the member_presenters override, I expect fragility, but I agree this is the right approach if the base code wasn't doing it all in one solr query and this makes it do so. :(